rfcs/0000-rlp-status.md

```diff
@@ -0,0 +1,157 @@
+* `Reconciled`: It communicates that the policy has successfully being reconciled, and the `ConfigMap` has been updated.
+* `Failed`: This one will be reached when either of `Validation` and `Reconcilation` processes have encounter any errors.
+
+## Stage 3
+**A final health check of services refines the status**
+
+The final stage will bring a greater degree of accuracy, thanks for a final process that will check the healthiness and 
+configuration version the `Limitador` service currently enforces.
+
+   ![](0000-rlp-status-assets/rlp_status_3.png)
+
+States rationale:
+
+* `Recondiled`: This state will precede the "Health check" process  graphed as `Service Probe` event.
+* `Protected`: After a successfull response of the `Service Probe`, this states communicates the policy is finally enforced.
+This is the final top-level condition.
+* `Failed`: Now this state could also be set after encountering errors in the `Service Probe` check.
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The stages before mentioned, will follow the [Kubernetes guidelines](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties) 
+regarding the Status object definition.
+
+
+**Conditions**
+
+| Type       | Status | Reason                | Message                                            | Top Level |
+|------------|--------|-----------------------|----------------------------------------------------|-----------|
+| Created    | True   | "PolicyCreated"       | "RateLimitPolicy created"                          | No        |
+| Updated    | True   | "PolicyUpdated"       | "RateLimitPolicy has been updated"                 | No        |
+| Applied    | True   | "PolicyApplied"       | "RateLimitPolicy has been successfully applied     | Yes       |
+|            | False  | "PolicyNotApplied"    | "RateLimitPolicy is invalid"                       |           |
+| Reconciled | True   | "PolicyReconciled"    | "RateLimitPolicy has been successfully reconciled" | Yes       |
+|            | False  | "PolicyNotReconciled" | "RateLimitPolicy failed reconciliation"            |           |
+| Protected  | True   | "PolicyEnforced"      | "RateLimitPolicy has been successfully enforced"   | Yes       |
+|            | False  | "PolicyNotEnforced"   | "RateLimitPolicy has encountered an error"         |           |
+
+
+A simplified version and more aligned with the Kubernetes objects implementation could be represented as the following.
+
+**Conditions**
+
+All conditions are top-level.
+
+| Type        | Status | Reason                      | Message                                                    |
+|-------------|--------|-----------------------------|------------------------------------------------------------|
+| Progressing | True   | "PolicyCreated"             | "RateLimitPolicy created"                                  |
```

Comment:
I like this simpler model. From this doc I feel it would read better.

Food for thought. I wondered if instead we need two types only

`protected` and `unprotected` ?